### ISSUE TITLE
Overfit Olan Modeller

### ISSUE THREAD

**Issue by ibaytar** [f1rzen/IST491#18]
Hoca ile konuştuğumda modellerin overfit olduğunu söyledi.

1. Lojistik regresyonda overfitten kurtulmalıyız. 
2. Ordered logistic regression olabilir. Sklearnden bakılması gerekiyor.
3. Eğer ordered olmuyorsa low-med birleştirilerek yapılabilir.
4. Logistic regression olmuyorsa naive bayes bakılabilir. Confussion matrixi daha tutarlı.
6. PowerBI gerek yok. Streamlit uygulaması devam edecek.
7. Streamlit uygulamasında ayrı bir sayfada dinamik grafik olacak.

**Comment by AshNumpy** [comment]
Lüften her bir konu için ayrı issue açalım.

1. Logistic regression'ın overfit olduğu konusu tartışılabilir. Test ve train verilerindeki başarıları arasında anlamlı bir fark görmüyorum.
     - ![image](https://github.com/f1rzen/IST491/assets/92748453/4d83ce1e-1917-436f-9dff-7fe6ddfe146d)

**NOT**
Eğer ortada bir sorun varsa ilgili soruna ilişkin issue açalım. Yapılacaklara ilişkin yorumları [milestone](https://github.com/f1rzen/IST491/milestones) olarak açalım.

**Comment by AshNumpy** [comment]
Overfit olmaya karşı Naive Bayes modelini kullanabiliriz. @f1rzen Senin de bu fikre katılman durumunda seçilen modeli naive Bayes olacak şekilde güncelleyeceğim.

![image](https://github.com/f1rzen/IST491/assets/92748453/1e72161c-7e98-4ddf-8b7f-118123ebe960)

* Model: Naive Bayes
* Train Accuracy: 0.88875
* Test Accuracy: 0.895
* Precision: 0.9076568592697626
* Recall: 0.8864099766538791

**Train Confusion matrix**
 [266   0  17]
 [  8 226  14]
 [ 50   0 219]

**Test Confusion matrix**
 [79  0  3]
 [ 2 47  6]
 [10  0 53]
